examples/incompressible-flow/3d-taylor-green-vortex/tgv-matrix-free.prm

```diff
@@ -0,0 +1,165 @@
+# Listing of Parameters
+#----------------------
+
+set dimension = 3
+
+#---------------------------------------------------
+# Simulation Control
+#---------------------------------------------------
+
+subsection simulation control
+  set method            = bdf3
+  set time step         = 0.025 # Time step
+  set number mesh adapt = 0     # If steady, nb mesh adaptation
+  set time end          = 20    # End time of simulation
+  set output name       = tgv   # Prefix for VTU outputs
+  set output frequency  = 2   # Frequency of simulation output
+  set output path       = ./output/
+  set subdivision       = 3
+end
+
+#---------------------------------------------------
+# FEM
+#---------------------------------------------------
+
+subsection FEM
+  set velocity order = 3
+  set pressure order = 3
+end
+
+#---------------------------------------------------
+# Timer
+#---------------------------------------------------
+
+subsection timer
+  set type = iteration # <none|iteration|end>
+end
+
+#---------------------------------------------------
+# Initial condition
+#---------------------------------------------------
+
+subsection initial conditions
+  set type = nodal
+  subsection uvwp
+    set Function expression = sin(x)*cos(y)*cos(z); -cos(x)*sin(y)*cos(z);0; 1./16*(cos(2*x)+cos(2*y))*(cos(2*z)+2);
+  end
+end
+
+#---------------------------------------------------
+# Physical Properties
+#---------------------------------------------------
+
+subsection physical properties
+  set number of fluids = 1
+  subsection fluid 0
+    set kinematic viscosity = 0.000625
+  end
+end
+
+#---------------------------------------------------
+# Post-Processing
+#---------------------------------------------------
+
+subsection post-processing
+  set verbosity                = verbose
+  set calculate enstrophy      = true
+  set calculate kinetic energy = true
+end
+
+#---------------------------------------------------
+# Mesh
+#---------------------------------------------------
+
+subsection mesh
+  set type               = dealii
+  set grid type          = hyper_rectangle
+  set grid arguments       = 1 : -3.14159265359 : 3.14159265359 : true
```

Comment:
Yeah I know. Initially I preferred to have the subdivided hyper_rectangle because then someone can easily have 3x3x3 or 5x5x5 to make non-multiple of 2 divisions, but both you and Laura seem to prefer the non-general version so let's go with the hypercube. I find that a bit sad though because if a user wants to make 96x cube or 48x cube now they need to go through all the documentation to figure it out.